Public/Src/Pips/Dll/Builders/ProcessBuilder.cs

```diff
@@ -103,6 +103,12 @@ public class ProcessBuilder : IDisposable
 
         private Dictionary<StringId, ProcessSemaphoreInfo> m_semaphores;
 
+        /// <summary>
+        /// The weighted value of this pip when limiting concurrency of process pips.
+        /// The higher the weight, the fewer process pips that can run in parallel.
+        /// </summary>
+        public int? Weight { get; set; } = null;

```

Comment:
>Weight [](start = 20, length = 6)

Should we error if this is negative or zero somewhere? I wonder if we should allow fractional values to represent lighter pips.

Reply:
We decided to just coerce < 1 to be 1. Because these represent process slots, I don't think anything < 1 is conceptually valid. 

Aleks said Dominoscript doesn't support fractional values, so it'd have to just be larger ints. 

We can use 0 in the future to indicate lightweight pips/pips that should just run regardless of whatever else is happening.


---
In reply to: [271499419](https://github.com/Microsoft/BuildXL/pull/67#discussion_r271499419) [](ancestors = 271499419)